zypp/RepoInfo.cc

```diff
@@ -186,7 +186,9 @@ namespace zypp
            && repo::RepoMirrorList::urlSupportsMirrorLink( *_baseUrls.transformedBegin() ) ) {
 
         mlurl = *_baseUrls.transformedBegin ();
-        mlurl.appendPathName("/");
+        if (!mlurl.getPathName().empty() && mlurl.getPathName() != "/" && !zypp::str::endsWith(mlurl.getPathName(), "/") ) {
```

Comment:
Silly comment prolly but the expression condition can also be catched by the last expression in the if (endswith slash returns true if the value if a lonely slash).

```suggestion
        if (!mlurl.getPathName().empty() && !zypp::str::endsWith(mlurl.getPathName(), "/") ) {
```